notify/opsgenie/opsgenie.go

```diff
@@ -211,13 +219,44 @@ func (n *Notifier) createRequests(ctx context.Context, as ...*types.Alert) ([]*h
 			responders = append(responders, responder)
 		}
 
+		var visibleTos []opsGenieCreateMessageVisibleTo
+		for _, v := range n.conf.VisibleTo {
+			visibleTo := opsGenieCreateMessageVisibleTo{
+				ID:       tmpl(v.ID),
+				Name:     tmpl(v.Name),
+				Username: tmpl(v.Username),
+				Type:     tmpl(v.Type),
+			}
+
+			if visibleTo == (opsGenieCreateMessageVisibleTo{}) {
+				// Filter out empty responders. This is useful if you want to fill
+				// responders dynamically from alert's common labels.
+				continue
+			}
+
+			if visibleTo.Type == "teams" {
+				teams := safeSplit(visibleTo.Name, ",")
+				for _, team := range teams {
+					newVisibleTo := opsGenieCreateMessageVisibleTo{
+						Name: tmpl(team),
```

Comment:
Why template the name a second time?

Reply:
Copy and paste from receivers, I did not catch that, thanks! Would you like me to remove the double templating in the code above too? Or have a dedicated PR for that? It does not change the results of the unit tests.

Fixed in new commit.